### PR TITLE
add e2e test

### PR DESCRIPTION
**Pull request by kaplanelad**
Create e2e test

# Checklist
- [x] Tests
- [x] Documentation
- [x] Linting

### CODE REVIEW

File: e2e/tests/put.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - path: settings/test/all
+    file_name: secret-a
+    content: mailman
+  - path: settings/test/all
+    file_name: secret-b
+    content: shazam
+  - path: settings/test/all
+    file_name: secret-c
+    content: shazam-1
+expected_snapshot:
```
Comment (jondot) [inline e2e/tests/put.yml:33]: What do you think about:
```
expected_snapshot:
   settings:
       test:
          foo: "new-foo-value"

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			}
+
+			if len(snapshot.ExpectedSnapshot) > 0 {
+				destSnapshotFolder := filepath.Join(tempFolder, "dest")
+				err = snapshot.CreateSnapshotData(snapshot.ExpectedSnapshot, destSnapshotFolder)
+				assert.Nil(t, err)
+
+				// consider to replace `diff` command which is depended on OS to golang plugin.
+				// could't find something better
+				s, _, err := testutils.ExecCmd("diff", []string{"-qr", "-x", snapshot.ConfigFileName, destSnapshotFolder, snapshotFolder}, "")
```
Comment (jondot) [inline e2e/e2e_test.go:80]: might want to encapsulate that under a method/interface that has a clear input/output.
that way when we come to find a different solution - we only need to implement to an interface
interface Differ{
   Diff(...)
}

**Comment by jondot** [comment]
Looks great. Not too much code and a lot of value!

BTW: is there a way to generate a new snapshot automatically as a starting point? (run teller in "recording" mode?)

**Review by jondot — APPROVED** [review]

File: e2e/e2e_test.go (hunk 2 above)
Comment (kaplanelad) [reply e2e/e2e_test.go]: totally agree, done

**Comment by kaplanelad** [comment]
> Looks great. Not too much code and a lot of value!
> 
> BTW: is there a way to generate a new snapshot automatically as a starting point? (run teller in "recording" mode?)

not yet, the recording mode is the next step :)

**Pull request merged**